reframe/frontend/executors/policies.py

```diff
@@ -394,20 +428,20 @@ def runcase(self, case):
 
                 return
 
-            if len(self._running_tasks[partname]) >= partition.max_jobs:
-                # Make sure that we still exceeded the job limit
-                getlogger().debug2(
-                    f'Reached concurrency limit for partition {partname!r}: '
-                    f'{partition.max_jobs} job(s)'
-                )
-                self._poll_tasks()
-
-            if len(self._running_tasks[partname]) < partition.max_jobs:
-                # Task was put in _ready_tasks during setup
-                self._ready_tasks[partname].pop()
-                self._reschedule(task)
-            else:
-                self.printer.status('HOLD', task.check.info(), just='right')
+            # if len(self._running_tasks[partname]) >= partition.max_jobs:
+            #     # Make sure that we still exceeded the job limit
+            #     getlogger().debug2(
+            #         f'Reached concurrency limit for partition {partname!r}: '
+            #         f'{partition.max_jobs} job(s)'
+            #     )
+            #     self._poll_tasks()
+
+            # if len(self._running_tasks[partname]) < partition.max_jobs:
+            #     # Task was put in _ready_tasks during setup
+            #     self._ready_tasks[partname].pop()
+            #     self._reschedule(task)
+            # else:
+            #     self.printer.status('HOLD', task.check.info(), just='right')
```

Comment:
Why did you comment that out?

Reply:
The problem is that it complicated the logic a bit - and unnecessarily in my opinion. For me it's simpler if we simply setup the test in the `runcase` and I commented this out to test it like this first. I am having a look if I can make it work with scheduling tasks as soon as possible.

Reply:
I think that we should schedule tasks as soon as possible. Imagine you have thousands of tests that need to load modules. By default the setup will try to track any conflicts by interacting with the module system, an operation that can be slow, so practically you'd have to pay this cost for all the tests before even actually trying to execute any test. Besides, the fact that we select to reschedule when we reach the partition concurrency limit is just a choice that has its root in reducing the poll ratio of the old, poll-hungry mechanism. We could poll at each an every call of `runcase()` which would also allow us to finalise tests as soon as possible, too. So I wouldn't eliminate this logic from here.